middlewares/pathReplace.go

```diff
@@ -0,0 +1,23 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+// PathReplace is a middleware used to replace the path of a URL request
+type PathReplace struct {
+	Handler http.Handler
+	Path    string
+}
+
+func (s *PathReplace) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.Header.Add("X-Replaced-Path", r.URL.Path)
+	r.URL.Path = s.Path
```

Comment:
A few questions:

1. Is it intentional that you replace the entire path?
1. Shouldn't we check if the path we intend to replace actually exists in `r.URL.Path` and only replace / add the header if it does?
1. What's the desired behavior if `s.Path` is shorter than `r.URL.Path`, i.e., we cannot replace it completely?